source/pkg/adapter/adapter.go

```diff
@@ -0,0 +1,351 @@
+			"noWait":   exchangeConfig.NoWait,
+		},)
+	if err != nil {
+		logger.Error(err.Error())
+		return nil, err
+	}
+
+	queue, err := (*ch).QueueDeclare(
+		a.config.QueueConfig.Name,
+		wabbit.Option{
+			"durable":   a.config.QueueConfig.Durable,
+			"delete":    a.config.QueueConfig.DeleteWhenUnused,
+			"exclusive": a.config.QueueConfig.Exclusive,
+			"noWait":    a.config.QueueConfig.NoWait,
+		},)
+	if err != nil {
+		logger.Error(err.Error())
+		return nil, err
+	}
+
+	if a.config.ExchangeConfig.TypeOf != "fanout" {
+		routingKeys := strings.Split(a.config.QueueConfig.RoutingKey, ",")
+
+		for _, routingKey := range routingKeys {
+			err = (*ch).QueueBind(
+				queue.Name(),
+				routingKey,
+				exchangeConfig.Name,
+				wabbit.Option{
+					"noWait": a.config.QueueConfig.NoWait,
+				},)
+			if err != nil {
+				logger.Error(err.Error())
+				return nil, err
+			}
+		}
+	} else {
+		err = (*ch).QueueBind(
+			queue.Name(),
+			"",
+			exchangeConfig.Name,
+			wabbit.Option{
+				"noWait": a.config.QueueConfig.NoWait,
+			})
+		if err != nil {
+			logger.Error(err.Error())
+			return nil, err
+		}
+	}
+
+	return &queue, nil
+}
+
+func (a *Adapter) ConsumeMessages(channel *wabbit.Channel,
+	queue *wabbit.Queue, logger *zap.Logger) (<-chan wabbit.Delivery, error) {
+	msgs, err := (*channel).Consume(
+		(*queue).Name(),
+		"",
+		wabbit.Option{
+			"noAck":     false,
+			"exclusive": a.config.QueueConfig.Exclusive,
+			"noLocal":   false,
+			"noWait":    a.config.QueueConfig.NoWait,
+		},)
+
+	if err != nil {
+		logger.Error(err.Error())
+	}
+	return msgs, err
+}
+
+func (a *Adapter) PollForMessages(channel *wabbit.Channel,
+	queue *wabbit.Queue, stopCh <-chan struct{}) error {
+	logger := a.logger
+
+	msgs, _ := a.ConsumeMessages(channel, queue, logger)
+
+	for {
+		select {
+		case msg, ok := <-msgs:
+			if ok {
+				logger.Info("Received: ", zap.Any("value", string(msg.Body())))
+
+				go func(message *wabbit.Delivery) {
+					if err := a.postMessage(*message); err == nil {
+						logger.Info("Successfully sent event to sink")
+						err = (*channel).Ack((*message).DeliveryTag(), true)
+						if err != nil {
+							logger.Error("Sending Ack failed with Delivery Tag")
+						}
+					} else {
+						logger.Error("Sending event to sink failed: ", zap.Error(err))
+						err = (*channel).Nack((*message).DeliveryTag(), true, true)
+						if err != nil {
+							logger.Error("Sending Nack failed with Delivery Tag")
+						}
+					}
+				}(&msg)
+			} else {
+				return nil
+			}
+		case <-stopCh:
+			logger.Info("Shutting down...")
+			return nil
+		}
+	}
+}
+
+func (a *Adapter) postMessage(msg wabbit.Delivery) error {
+	a.logger.Info("url ->" + a.httpMessageSender.Target)
+	req, err := a.httpMessageSender.NewCloudEventRequest(a.context)
+	if err != nil {
+		return err
+	}
+
+	event := cloudevents.NewEvent()
+
+	event.SetID(msg.MessageId())
+	event.SetTime(msg.Timestamp())
+	event.SetType(sourcesv1alpha1.RabbitmqEventType)
```

Comment:
Maybe not in this initial PR, but there would ideally be a way for a user header to specify the event type (since rabbitmq is typically just going to be the transport, not the "type"). This might lead to a discussion of a CloudEvent binding for amqp 0.91.